server/api/main.py

```diff
@@ -289,18 +292,35 @@ async def _verify_log_collection_started_on_startup(
             states=mlrun.runtimes.constants.RunStates.terminal_states(),
         )
     )
-    if runs:
-        logger.debug(
-            "Found runs which require logs collection on startup",
-            runs_uids=[run.get("metadata", {}).get("uid", None) for run in runs],
-        )
+    if runs_uids:
+        if len(runs_uids) <= config.log_collector.start_logs_startup_run_limit:
```

Comment:
You may need to cast it. Also mlconf as convention
```suggestion
        if len(runs_uids) <= int(mlrun.mlconf.log_collector.start_logs_startup_run_limit):
```